app/scenes/Settings/Security.tsx

```diff
@@ -36,9 +36,22 @@ function Security() {
     defaultUserRole: team.defaultUserRole,
     memberCollectionCreate: team.memberCollectionCreate,
     inviteRequired: team.inviteRequired,
-    allowedDomains: team.allowedDomains,
   });
 
+  const [allowedDomains, setAllowedDomains] = useState([
+    ...(team.allowedDomains ?? []),
+  ]);
+  const [lastKnownDomainCount, updateLastKnownDomainCount] = useState(
+    allowedDomains.length
+  );
+
+  const [existingDomainsTouched, setExistingDomainsTouched] = useState(false);
+
+  const showSaveDomainsButton = React.useCallback(() => {
```

Comment:
I'll be honest, I'm not hugely familiar with React Hooks so I wasn't sure about this. I put it in because it performs an array manipulation so I thought it _could_ become expensive (because running `Array.filter` creates a copy). Then again, it's unlikely to be a big array, and I'm happy to defer to your judgment on how expensive this memoisation is.
Let me know if you still want me to remove it and I'll do it!

Reply:
Yep, just calculate the boolean inline – it's not a problem here 👍